handsontable/src/plugins/copyPaste/clipboardData/clipboardData.js

```diff
@@ -0,0 +1,558 @@
+      of elements (corresponding to size of the dataset in other columns, including headers) for inserted columns.`;
+    }
+  }
+
+  /**
+   * Adjust information about merged cells after column insertion.
+   *
+   * Note: Used index refers to processed data, not to the instance of Handsontable.
+   *
+   * @private
+   * @param {object} metaInfo Object containing `data`, `colHeaders`, `rowHeaders`, `nestedHeaders`, `mergeCells`
+   * keys and the corresponding values, which will be changed by the reference.
+   * @param {number} columnIndex An index of the column at which the new values have been inserted.
+   */
+  adjustAfterColumnInsertion(metaInfo, columnIndex) {
+    const { mergeCells: mergedCells, data } = metaInfo;
+
+    mergedCells?.forEach((mergeArea) => {
+      const { row: mergeStartRow, col: mergeStartColumn, colspan, rowspan } = mergeArea;
+
+      if (columnIndex > mergeStartColumn && columnIndex < mergeStartColumn + colspan) {
+        mergeArea.colspan += 1;
+
+        for (let i = 0; i < rowspan; i += 1) {
+          data[mergeStartRow + i][columnIndex] = '';
+        }
+      }
+    });
+
+    mergedCells?.forEach((mergeArea) => {
+      if (columnIndex <= mergeArea.col) {
+        mergeArea.col += 1;
+      }
+    });
+  }
+
+  /**
+   * Insert values at column index.
+   *
+   * Note: Used index refers to processed data, not to the instance of Handsontable.
+   *
+   * @param {number} columnIndex An index of the column at which the new values will be inserted or removed.
+   * @param {string[]} values List of values.
+   */
+  insertAtColumn(columnIndex, values) {
+    if (this.isSerializedTable() === false) {
+      return;
+    }
+
+    const metaInfo = this.getMetaInfo();
+    const { data, colHeaders } = metaInfo;
+    const headerLevels = isDefined(colHeaders) ? 1 : 0;
+    const columnInsertionWarn = this.getColumnInsertionWarn(columnIndex, values);
+
+    if (isDefined(columnInsertionWarn)) {
+      warn(columnInsertionWarn);
+
+      return;
+    }
+
+    if (headerLevels > 0) {
+      colHeaders.splice(columnIndex, 0, values[0]);
+    }
+
+    data?.forEach((rowData, rowIndex) => {
+      rowData.splice(columnIndex, 0, values[rowIndex + headerLevels]);
+    });
+
+    this.adjustAfterColumnInsertion(metaInfo, columnIndex);
+    this.overwriteInfo(metaInfo);
+  }
+
+  /**
+   * Change headers or cells in the serialized dataset.
+   *
+   * Note: Used indexes refers to processed data, not to the instance of Handsontable. Please keep in mind that headers
+   * are handled separately from cells and they are recognised using negative indexes.
+   *
+   * @param {number} row Row index of cell which should be changed.
+   * @param {number} column Column index of cell which should be changed.
+   * @param {string} value Value for particular indexes.
+   * @param {boolean} isRtl Grid is rendered using the right-to-left layout direction.
+   */
+  setCellAt(row, column, value, isRtl = false) {
```

Comment:
```suggestion
  setCellAt(row, column, value) {
```
I think having the 4th argument (`isRtl`) here is a bad idea. Shouldn't it be that this value is determined by the table settings? So the code internally would call `hot.isRtl()`?

Reply:
It has been removed.